From b49bc7619031cd35eb3b90484b7f8716eba91887 Mon Sep 17 00:00:00 2001
From: Nicholas Kuechler <nkuechler@gmail.com>
Date: Thu, 27 Jun 2024 13:57:29 -0500
Subject: [PATCH 22/22] horizon: fix templating of list of strings for
 CSRF_TRUSTED_ORIGINS

Change-Id: I740cd48103950e1599e77db46c7e4d9e65677177

[ Cherry-picked to stx-openstack caracal ]
Test Plan:
[PASS] build stx-openstack tarball
[PASS] install openstack with certificate and domain_name setup, enabling HTTPs
[PASS] Access horizon via fqdn (ex: https://horizon-<domain>.com/)
Related-Bug: #2103799
Signed-off-by: Alex Figueiredo <alex.fernandesfigueiredo@windriver.com>
---
 horizon/values.yaml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/horizon/values.yaml b/horizon/values.yaml
index 6362ad87..fbf465f5 100644
--- a/horizon/values.yaml
+++ b/horizon/values.yaml
@@ -314,7 +314,7 @@ conf:
         SESSION_COOKIE_HTTPONLY = {{ .Values.conf.horizon.local_settings.config.session_cookie_httponly }}
 
         # https://docs.djangoproject.com/en/dev/ref/settings/#csrf-trusted-origins
-        CSRF_TRUSTED_ORIGINS = {{ .Values.conf.horizon.local_settings.config.csrf_trusted_origins }}
+        CSRF_TRUSTED_ORIGINS = [{{ include "helm-toolkit.utils.joinListWithCommaAndSingleQuotes" .Values.conf.horizon.local_settings.config.csrf_trusted_origins }}]
 
         # Overrides for OpenStack API versions. Use this setting to force the
         # OpenStack dashboard to use a specific API version for a given service API.
-- 
2.34.1

